dali/operators/reader/video_reader_decoder_op_test.cc

```diff
@@ -166,6 +167,88 @@ TEST_F(VideoReaderDecoderCpuTest, CpuVariableFrameRate_CpuOnlyTests) {
   }
 }
 
+TEST_F(VideoReaderDecoderGpuTest, GpuVariableFrameRate) {
+  const int batch_size = 4;
+  const int sequence_length = 6;
+  const int stride = 3;
+  const int step = 10;
+
+  Pipeline pipe(batch_size, 4, 0);
+
+  pipe.AddOperator(OpSpec("experimental__readers__Video")
+    .AddArg("device", "gpu")
+    .AddArg("sequence_length", sequence_length)
+    .AddArg("stride", stride)
+    .AddArg("step", step)
+    .AddArg(
+      "filenames",
+      std::vector<std::string>{
+        testing::dali_extra_path() + "/db/video/vfr/test_1.mp4",
+        testing::dali_extra_path() + "/db/video/vfr/test_2.mp4"})
+    .AddArg("labels", std::vector<int>{0, 1})
+    .AddArg("initial_fill", 1)
+    .AddArg("lazy_init", true)
+    .AddOutput("frames", "gpu")
+    .AddOutput("labels", "gpu"));
+
+  pipe.Build({{"frames", "gpu"}, {"labels", "gpu"}});
+  // pipe.Build({{"frames", "gpu"}});
+
+  int num_sequences = 20;
+  int sequence_id = 0;
+  int batch_id = 0;
+  int gt_frame_id = 0;
+
+  int video_idx = 0;
+
+  std::vector<uint8_t> frame_cpu(std::max(
+    this->FrameSize(0), this->FrameSize(1)));
+
+  while (sequence_id < num_sequences) {
+    DeviceWorkspace ws;
+    pipe.RunCPU();
+    pipe.RunGPU();
+    pipe.Outputs(&ws);
+
+    auto &frame_video_output = ws.template Output<dali::GPUBackend>(0);
+    auto &frame_label_output = ws.template Output<dali::GPUBackend>(1);
+
+    ASSERT_EQ(frame_video_output.GetLayout(), "NFHWC");
```

Comment:
```suggestion
    ASSERT_EQ(frame_video_output.GetLayout(), "FHWC");
```
I think "F" is implicit.

Reply:
Done